nim: 0.19.4 -> 0.20.0

royneary opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#sec-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change
Update to the latest release of Nim.

What I did:
- ~~makeWrapper not needed anymore~~
- link sqlite (needed for the the stdlib sqlite module)
- update nodejs to 11.x
- use default phase order (fixes linker errors during test phase)
- substitution in ./test/async/tioselectors.nim not needed anymore
- two more tests need to be disabled
- LD=$CC workaround not needed anymore
- update mosdepth for it to compile with nim 0.20.0

nrpl compiles with the new version.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

mmahut reviewed (approved):
- [X] reviewed the diff and commit messages
- [X] made sure ofBorg evals
- [X] run nix-review without any failures
- [X] run and tested the binaries

veprbl commented:
@GrahamcOfBorg build nim mosdepth nrpl

veprbl commented:
cc @ehmry @peterhoeg

royneary commented:
Some tests [lack support for aarch64](https://github.com/nim-lang/Nim/issues/11463). I disabled them for the architecture.

peterhoeg commented:
Nice cleanups as well! Could you please run the relevant ```pre``` and ```post``` hooks from the various phases as well?

Also, are you sure that ```tzdata``` isn't needed a proper buildInput?

royneary commented:
I'm triggering the pre and post hooks now (and don't override patchPhase anymore).

I think tzdata was only added as a dependency to fix the path ``/usr/share/zoneinfo`` in a test. So it's only needed during the checkPhase and not as a buildInput.

veprbl commented:
> I think tzdata was only added as a dependency to fix the path `/usr/share/zoneinfo` in the test case `./tests/stdlib/ttimes.nim`. So it's only needed during the checkPhase and not as a buildInput.

You mean this?
https://github.com/NixOS/nixpkgs/blob/567a05f6ab1ae35ffc076959aed188e27a74b190/pkgs/development/compilers/nim/default.nix#L69

In that case we could drop it from `nativeBuildInputs`. Or otherwise move to `checkInputs`.

royneary commented:
Oh, I didn't know ``checkInputs``. The same applies for ``nodejs-slim-11_x`` and ``coreutils`` then.

veprbl commented:
@royneary Could you please squash the nim-related commits?
Also might want to fix your commit author email. You could use "royneary@users.noreply.github.com" if you don't want to share your personal address.

royneary commented:
@veprbl I squashed the commits. Are you suggesting to change my email address because github wasn't able to verify my GPG signatures? I fixed that.

veprbl commented:
@royneary Thanks! I was mostly worried about the commit attribution to your username.

There is an eval error now. I guess you missing some parenthesis.

royneary commented:
@veprbl fixed. It was because in master someone removed variable that was then unused but now used again :-)

veprbl commented:
@GrahamcOfBorg build nim mosdepth nrpl

veprbl commented on pkgs/development/compilers/nim/default.nix:
```diff
@@ -48,35 +47,53 @@ stdenv.mkDerivation rec {
-    mv $out/nim/bin/* $out/bin/ && rmdir $out/nim/bin
-    mv $out/nim/*     $out/     && rmdir $out/nim
-    wrapProgram $out/bin/nim \
-      --suffix PATH : ${lib.makeBinPath [ stdenv.cc ]}
```
This wrapper is still needed, I think.

veprbl replied:
Does "nim compile" work for you in pure environment?

royneary replied:
No, it doesnt, you are right. I fixed it.

royneary committed:
mosdepth: 0.2.3 -> 0.2.6

royneary committed:
nim: 0.19.4 -> 0.20.0

- link sqlite (needed for the the stdlib sqlite module)
- update nodejs to 11.x
- use default phase order (fixes linker errors during test phase)
- substitution in ./test/async/tioselectors.nim not needed anymore
- two more tests need to be disabled
- LD=$CC workaround not needed anymore
- disable unsupported tests on aarch64
- trigger pre and post hooks
- use checkInputs instead of nativeBuildInputs
- don't override patchPhase

veprbl commented:
@GrahamcOfBorg build nim mosdepth nrpl
